js/iframe-coordinator/libs/workers/worker-manager.ts

```diff
@@ -0,0 +1,288 @@
+    WorkerManager.trackedWorkerEventTypes.forEach(curr => {
+      worker.addEventListener(curr, this);
+    });
+
+    this._workers.push({
+      id,
+      url,
+      worker,
+      phase: WorkerPhase.LOADING,
+      errorWindowCount: 0,
+      errorWindowStartTimestamp: -1
+    });
+
+    return id;
+  }
+
+  public stop(idToStop: string) {
+    const workerIndexToStop = this._workers.findIndex((curr, index) => {
+      return curr.id === idToStop;
+    });
+
+    if (workerIndexToStop >= 0) {
+      const worker = this._workers[workerIndexToStop].worker;
+      WorkerManager.trackedWorkerEventTypes.forEach(curr => {
+        worker.removeEventListener(curr, this);
+      });
+      worker.terminate();
+
+      this._workers.splice(workerIndexToStop, 1);
+    }
+  }
+
+  // TODO Need comms to workers (pub/sub, others?)
+
+  // Event Management
+  public addEventListener(type: string, listener: any, capture?: any): void {
+    WorkerManager.evtTargetDelegate.addEventListener(type, listener, capture);
+  }
+
+  public dispatchEvent(event: Event): boolean {
+    return WorkerManager.evtTargetDelegate.dispatchEvent(event);
+  }
+
+  public removeEventListener(type: string, listener: any, capture?: any): void {
+    WorkerManager.evtTargetDelegate.removeEventListener(
+      type,
+      listener,
+      capture
+    );
+  }
+
+  public handleEvent(evt: Event): void {
+    if (evt instanceof MessageEvent) {
+      this._onWorkerMsg(evt);
+    } else if (evt instanceof ErrorEvent) {
+      const managedWorker = this._workers.find(curr => {
+        return curr.worker === evt.target;
+      });
+
+      if (managedWorker) {
+        const currentTimestamp = Date.now();
+
+        if (managedWorker.errorWindowStartTimestamp < 0) {
+          // First Error
+          managedWorker.errorWindowStartTimestamp = currentTimestamp;
+          managedWorker.errorWindowCount = 1;
+          return;
+        }
+
+        const delta = Math.max(
+          currentTimestamp - managedWorker.errorWindowStartTimestamp
+        );
+        if (delta > this._errorWindowMillis) {
+          // Window expired; start new window
+          managedWorker.errorWindowStartTimestamp = currentTimestamp;
+          managedWorker.errorWindowCount = 1;
+          return;
+        } else {
+          managedWorker.errorWindowCount++;
+
+          if (
+            managedWorker.errorWindowCount > this._errorWindowCountThreshold
+          ) {
+            // TODO Need to add proper logging support
+            // tslint:disable-next-line
+            console.error('Error rate exceeded.  Stopping the worker', {
+              workerDetails: managedWorker,
+              errorWindowCount: managedWorker.errorWindowCount,
+              errorWindowStartTimestamp:
+                managedWorker.errorWindowStartTimestamp,
+              errorWindowDurationMillis: delta,
+              errorDetails: evt.error
+            });
+            this.stop(managedWorker.id);
```

Comment:
I decided on a configurable error rate.  If exceeded, we will log and stop the worker.  We can tune this as needed.

Also, we could allow the client to configure a restart or allow the register function to be recalled.  Thoughts?